dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1259,29 +1259,32 @@ class IndexConsumer : public ASTConsumer,
       StringRef relativePath,
       const Module *imported) {
     PresumedLoc presumedHashLoc = sm.getPresumedLoc(hashLoc);
-    const FileInfoPtr &source = getFileInfo(presumedHashLoc.getFilename());
-    const FileInfoPtr &target = getFileInfo(file->getName());
-    SourceLocation targetBegin, targetEnd;
-
     if (!interestingLocation(hashLoc) ||
         filenameRange.isInvalid() ||
-        presumedHashLoc.isInvalid() ||
+        presumedHashLoc.isInvalid())
+      return;
 
-        // Don't record inclusions of files that are outside the source tree,
-        // like stdlibs. file is NULL if an #include can't be resolved, like if
-        // you include a nonexistent file.
-        !file ||
+    // Don't record inclusions of files that are outside the source tree,
+    // like stdlibs. file is NULL if an #include can't be resolved, like if
+    // you include a nonexistent file.
+    if (!file)
+      return;
 
-        !(target->interesting) ||
+    const FileInfoPtr &source = getFileInfo(presumedHashLoc.getFilename());
```

Comment:
I guess we could even move this down a few more lines to nearer where it's used.